src/components/dropdown-button/dropdown-button.tsx

```diff
@@ -0,0 +1,262 @@
+  @State() filteredOptions = [];
+
+  /**
+   * State representing dropdown is open or not
+   */
+  @State() isDropdownOpen = false;
+
+  /**
+   * Triggered when an option is clicked
+   */
+  @Event() fwOptionClick: EventEmitter<any>;
+
+  /**
+   * Triggered when Add button for searchable dropdown is clicked
+   */
+  @Event() fwOptionsAdd: EventEmitter<any>;
+
+  @Watch('isDropdownOpen')
+  setInitialState(newVal) {
+    // Do not open dropdown if options is absent
+    if (newVal) {
+      if (this.options.length === 0) {
+        this.isDropdownOpen = false;
+      }
+    }
+    // Reset searchable dropdown to initial state on close
+    if (!newVal) {
+      if (this.searchable) {
+        this.filteredOptions = this.options;
+        this.dropdownInput.value = '';
+      }
+    }
+  }
+
+  /**
+   * Toggles dropdown b/w open and close
+   */
+  private handleDropdownToggle() {
+    this.isDropdownOpen = !this.isDropdownOpen;
+  }
+
+  /**
+   * Sets dropdown options from slot
+   */
+  private setDropdownOptions() {
+    // Set options if the prop is not supplied by user
+    if (this.options.length === 0) {
+      this.options = Array.from(
+        this.host.querySelectorAll('[slot=dropdown-options] option')
+      ).map(option => ({
+        id: option.id,
+        value: option.getAttribute('value'),
+        label: option.textContent,
+      }));
+    }
+
+    // Set filtered options to inital options for search dropdown
+    if (this.searchable) {
+      this.filteredOptions = this.options;
+    }
+  }
+
+  /**
+   * Sets search dropdown input
+   */
+  private setSearchInput() {
+    this.optionInput = this.dropdownInput.value;
+    this.filteredOptions = this.optionInput !== ''
+      ? this.options.filter(option => option.label.toLowerCase()
+        .includes(this.optionInput.toLowerCase()))
+      : this.options;
+  }
+
+  /**
+   * Listens to checkbox change for search dropdown
+   */
+  private handleCheckboxChange(ev) {
+    if (ev.target.checked) {
+      this.value = [
+        ...(this.value || []),
+        this.options.find(({ value }) => value === ev.target.id).value,
+      ];
+    } else {
+      this.value = this.value.filter(val => val !== ev.target.id);
+    }
+  }
+
+  /**
+   * Handles `Add` button click in search dropdown
+   */
+  private handleAddClick() {
+    this.fwOptionsAdd.emit({ value: this.value });
+    this.handleDropdownToggle();
+  }
+
+  /**
+   * Handles option select from the dropdown
+   */
+  private handleOptionClick(option) {
+    if (!this.searchable) {
+      this.value = option.value;
+      this.fwOptionClick.emit({ value: this.value });
+      this.handleDropdownToggle();
+    }
+  }
+
+  render() {
+    const ChevronDown = () => {
+      const iconColor = ['secondary', 'link', 'text'].includes(this.color) ? '#12344d' : '#fff';
+      const direction = this.isDropdownOpen ? 'up' : 'down';
+
+      return <fw-icon name={`chevron-${direction}`} color={iconColor} size="8" > </fw-icon>;
```

Comment:
Revert `fw-icon` size to a number. You can add it as a number here.

Reply:
I see an error for number. Anyways will check it and revert!